src/rdf4cpp/IRIFactory.cpp

```diff
@@ -226,6 +229,14 @@ nonstd::expected<IRI, IRIFactoryError> IRIFactory::create_and_validate(std::stri
     return IRI::make_unchecked(iri, node_storage);
 }
 
+IRIFactoryError IRIFactory::assign_prefix_checked(std::string_view prefix, std::string_view expanded) {
+    using namespace util::char_matcher_detail;
```

Comment:
please add a comment inside the function that we do not need to check the `expanded` because it will be done on IRI creation